### PR TITLE
List of teachers

### PR DESCRIPTION
**Pull request by Tarkeasy**
1 .Added a function to add new teacher
2. Added form validation
Changing the existing teacher does not work due to problems on the server, once the server issues are resolved, this function will be working

### CODE REVIEW

File: src/app/admin-panel/admin-panel/teachers/teachers.component.html [hunk 1]
```diff
@@ -1,3 +1,107 @@
-<p>
-  teachers works!
-</p>
+<p-table [columns]="columns" [value]="teachers" selectionMode="single" [(selection)]="selectedTeacher" (onRowSelect)="onRowSelect($event)"
+    [paginator]="false" [rows]="50" >
+    <ng-template pTemplate="caption">
+        Список учителів
+    </ng-template>
+    <ng-template pTemplate="header" let-columns>
+        <tr>
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/teachers/teachers.component.html]: indentation

File: src/app/admin-panel/admin-panel/teachers/teachers.component.ts [hunk 2]
```diff
@@ -1,15 +1,75 @@
...
+      .getTeachers()
+      .subscribe(users => (this.teachers = users));
+    this.columns = [
+      { field: "firstname", header: "Ім'я" },
+      { field: "lastname", header: "Прізвище" },
+      { field: "patronymic", header: "По батькові" },
+      { field: "dateOfBirth", header: "Дата народження" }
+    ];
+  }
+  handlerFileInput(file: FileList){
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/teachers/teachers.component.ts]: run linter

File: src/app/admin-panel/admin-panel/teachers/teachers.component.html (hunk 1 above)
Comment (Tarkeasy) [reply src/app/admin-panel/admin-panel/teachers/teachers.component.html]: > indentation

fixed

File: src/app/admin-panel/admin-panel/teachers/teachers.component.ts (hunk 2 above)
Comment (Tarkeasy) [reply src/app/admin-panel/admin-panel/teachers/teachers.component.ts]: > run linter

fixed

**Pull request merged**